Refactor skills

johnnypowa opened the pull request:
### Refactor skills

-AppDatabase version change 1 to 2
-Create typeConverter to skillList
-Add skills columnInfo to KnightEntity
-Add skillList to Knight
-Create skills in CombatLocalDataSource
-Add skills to skillList in CombatLocalDataSource
-Add skillList to KnightDao.insertKnight in CombatLocalDataSource
-Create skillActivation function and checkEquippedSkills function, remove old skill functions on CombatFragment
-Use skillActivation function on knightSkills setOnClickListeners
-Add skillList to knightEntityData on CombatViewModel
-Create Skill class

### Plan

-Show the equipped skills icons and names on the CombatFragment
-Create RecyclerView layout for skills 
-Users can equip or unequip the skills on the skills fragment. Max. 4 skills can be equipped.

johnnypowa committed:
Refactor skills

-AppDatabase version change 1 to 2
-Create typeConverter to skillList
-Add skills columnInfo to KnightEntity
-Add skillList to Knight
-Create skills in CombatLocalDataSource
-Add skills to skillList in CombatLocalDataSource
-Add skillList to KnightDao.insertKnight in CombatLocalDataSource
-Create skillActivation function and checkEquippedSkills function, remove old skill functions on CombatFragment
-Use skillActivation function on knightSkills setOnClickListeners
-Add skillList to knightEntityData on CombatViewModel
-Create Skill class

csepidave commented on app/src/main/java/com/example/tanuls2/ui/view/CombatFragment.kt:
```diff
@@ -311,51 +316,57 @@ class CombatFragment : Fragment() {
-        skillDisable(knightSkill3Id)
-        printKnightParameter(combatViewModel.currentKnight!!)
-        printZombieParameter(zombie)
+    fun checkEquippedSkills(){
```
let's talk about it :speech_balloon:

johnnypowa replied:
Done

csepidave commented on app/src/main/java/com/example/tanuls2/ui/view/CombatFragment.kt:
```diff
@@ -311,51 +316,57 @@ class CombatFragment : Fragment() {
     }
 
-    fun skillBloodSiphon() {
+    fun skillActivation(index : Int){
```
let's talk about it :speech_balloon:

johnnypowa replied:
Done

csepidave commented on app/src/main/java/com/example/tanuls2/service/datasource/CombatLocalDataSource.kt:
```diff
@@ -3,24 +3,29 @@ package com.example.tanuls2.service.datasource
         return Single.fromCallable {
             if(SharedPreferencesHandler.isFirstStart()){
-                var itemList = arrayListOf<Item>()
+                var skillList = arrayListOf(skillExtraDamage,skillCriticalHit,skillPrecisionHit,skillLifeSteal,skillHeal)
```
segmentation!
```
arrayListOf(skillExtraDamage, skillCriticalHit, skillPrecisionHit, skillLifeSteal, skillHeal)
```

johnnypowa replied:
Done

csepidave commented on app/src/main/java/com/example/tanuls2/service/datasource/CombatLocalDataSource.kt:
```diff
@@ -3,24 +3,29 @@ package com.example.tanuls2.service.datasource
 
 class CombatLocalDataSource(val knightDao: KnightDao) {
 
+    var skillExtraDamage = Skill("Dupla ütés", 1,0,100,0f,0f,0f,true)
```
Please pay attention to segmentation!
```
Skill("Dupla ütés", 1, 0, 100, 0f, 0f , 0f, true)
```

johnnypowa replied:
Done

csepidave commented on app/src/main/java/com/example/tanuls2/model/Skill.kt:
```diff
@@ -0,0 +1,11 @@
+                 var level:Int,
+                 var health:Int,
+                 var damage:Int,
+                 var criticalHitChance: Float,
```
Use the same segmentation like this on the other lines too.

johnnypowa replied:
Done

csepidave commented on app/src/main/java/com/example/tanuls2/ui/view/CombatFragment.kt:
```diff
@@ -311,51 +316,57 @@ class CombatFragment : Fragment() {
-            true -> {
-                combatViewModel.currentKnight!!.maxHealth
+        when(equippedSkillsList[index].skillName){
+            getString(R.string.skill_double_hit)->{
```
segmentation

johnnypowa replied:
Done

csepidave reviewed (changes requested):
There are some small formatting issues and issues which we have to discuss.
But overall nice work! 👍

johnnypowa committed:
Skills refactor fix PR issues

- Create enums for Skill class, rename skillName to skillLabel
- Add constans instead of hard code for skillLabel
- Simplify checkEquippedSkills function
- Use enums
